src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx

```diff
@@ -52,35 +61,69 @@ function AnalyticalStatementInput(props: AnalyticalStatementInputProps) {
         onValueRemove: onAnalyticalEntryRemove,
     } = useFormArray('analyticalEntries', value.analyticalEntries ?? [], onFieldChange);
 
-    const handleAnalyticalEntryAdd = useCallback(
-        (val: Record<string, unknown> | undefined) => {
-            if (!val) {
-                return;
+    const handleAnalyticalEntryDrop = useCallback(
+        (dropValue: DroppedValue, dropOverEntryUuid?: string) => {
+            if (isDefined(dropValue.statementUuid) && (dropValue.statementUuid !== value.uuid)) {
+                // NOTE: Remove entry from old statement if it was moved from
+                // one statement to another statement
+                console.warn('i am being removed', dropValue);
             }
-            const typedVal = val as { entryId: number };
+            const newAnalyticalEntries = [...(value.analyticalEntries ?? [])];
 
-            // NOTE: Don't add what is already added
-            if (value.analyticalEntries?.find(item => item.entry === typedVal.entryId)) {
-                return;
+            const uuid = randomString();
+            let newAnalyticalEntry: PartialForm<AnalyticalEntryType> = {
+                uuid,
+                entry: dropValue.entryId,
+                order: value.order ?? 0,
+            };
+
+            // NOTE: Treat moved item as a new item by removing the old one and
+            // adding the new one again
+            const movedItem = value.analyticalEntries
+                ?.find(item => item.entry === dropValue.entryId);
+            if (value.analyticalEntries && isDefined(movedItem)) {
+                newAnalyticalEntry = {
+                    ...movedItem,
+                    order: value.order ?? 0,
+                };
+                const movedItemOldIndex = value.analyticalEntries
+                    .findIndex(item => item.entry === dropValue.entryId);
+                newAnalyticalEntries.splice(movedItemOldIndex, 1);
             }
+
             // NOTE: Don't let users add more that certain entries
-            if ((value.analyticalEntries?.length ?? 0) >= ENTRIES_LIMIT) {
+            if (
+                isNotDefined(movedItem)
+                && (value?.analyticalEntries?.length ?? 0) >= ENTRIES_LIMIT
+            ) {
                 return;
             }
```

Comment:
We should move this before `const newAnalyticalEntries = [...(value.analyticalEntries ?? [])];`